### PR TITLE
Mak/improve routing

### LINKED ISSUE #98: Refactor routes and organise server project structure
We load _all_ of the routes into `server/index.ts` currently and it's getting quite cluttered, we can split a lot of this out into separate routes

### PR DESCRIPTION
**Pull request by mordamax**
Fix #98

### CODE REVIEW

**Review by joao-paulo-parity — COMMENTED** [review]
FWIW here's my intuition for how to organize the modules. I still need to open the ticket for explaining how to go about this more-or-less, but it might be helpful for now.

Current:

```
src/server
├── checkHealth.ts
├── constants.ts
├── index.ts
├── lib
│  ├── Actions.ts
│  ├── ErrorCounter.ts
│  ├── rpcApiInstance.ts
│  └── Storage.ts
└── router
   ├── actions.ts
   ├── healthcheck.ts
   ├── index.ts
   └── metrics.ts
```

**Starting out**, I would aim for this kind of structure:

```
src/server
├── constants.ts
├── main.ts
└── routes
   ├── actions.ts
   ├── healthcheck.ts
   ├── index.ts
   └── metrics.ts
```

After that I would try to figure out where the `lib` modules should be. For instance, `lib/Storage` is unclear to me. Is it a storage of what? The only place it's imported is in the `router/actions` file, so is it an "actions storage"? If so, it could be like this:

``` 
src/server
├── constants.ts
├── actions
   └── storage.ts
└── routes
   └── actions.ts
```

Same for `ErrorCounter`: is it counting what kinds of errors? "Action errors"? If so it could be put in a `server/actions/ErrorCounter` module, an so on.

**Review by chevdor — APPROVED** [review]

File: src/server/checkHealth.ts [hunk 1]
```diff
@@ -1,14 +1,14 @@
 import { Request, Response } from 'express';
 
 import { logger } from '../utils';
-import apiInstance from './rpc';
+import rpcApiInstance from './lib/rpcApiInstance';
```
[the remark this answers is not in the document]
Comment (mordamax) [reply src/server/checkHealth.ts]: I'm not sure if that bad idea to introduce something like this on some projects where it make sense
for example here - whether it's `lib` or `services` - this is a folder with reusable or potentially reusable classes. 
These are not "utils" helpers, but standalone functional or domain classes, which can be grouped together

File: src/server/lib/rpcApiInstance.ts [hunk 2]
```diff
@@ -12,6 +12,6 @@ const injectedTypes = JSON.parse(
 
 const provider = new HttpProvider(rpcEndpointUrl);
 const types = injectedTypes;
-const apiInstance = new ApiPromise({ provider, types });
+const rpcApiInstance = new ApiPromise({ provider, types });
 
-export default apiInstance;
+export default rpcApiInstance;
```
[the remark this answers is not in the document]
Comment (mordamax) [reply src/server/lib/rpcApiInstance.ts]: [c2911b5](https://github.com/paritytech/substrate-matrix-faucet/pull/145/commits/c2911b5f98cd03eea7543d1e7efea40dd33739e9)

File: src/server/router/index.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import express from 'express';
+
+import actionRoutes from './actions';
```
[the remark this answers is not in the document]
Comment (mordamax) [reply src/server/router/index.ts]: [c2911b5](https://github.com/paritytech/substrate-matrix-faucet/pull/145/commits/c2911b5f98cd03eea7543d1e7efea40dd33739e9)

**Review by joao-paulo-parity — APPROVED** [review]

**Pull request merged**